discord/sku.py

```diff
@@ -0,0 +1,175 @@
+
+    __slots__ = (
+        '_state',
+        'id',
+        'type',
+        'application_id',
+        'name',
+        'slug',
+        '_flags',
+    )
+
+    def __init__(self, *, state: ConnectionState, data: SKUPayload):
+        self._state: ConnectionState = state
+        self.id: int = int(data['id'])
+        self.type: SKUType = try_enum(SKUType, data['type'])
+        self.application_id: int = int(data['application_id'])
+        self.name: str = data['name']
+        self.slug: str = data['slug']
+        self._flags: int = data['flags']
+
+    def __repr__(self) -> str:
+        return f'<SKU id={self.id} name={self.name!r} slug={self.slug!r}>'
+
+    @property
+    def flags(self) -> SKUFlags:
+        """Returns the flags of the SKU."""
+        return SKUFlags._from_value(self._flags)
+
+    @property
+    def created_at(self) -> datetime:
+        """:class:`datetime.datetime`: Returns the sku's creation time in UTC."""
+        return utils.snowflake_time(self.id)
+
+
+class Entitlement:
+    """Represents an entitlement from user or guild which has been granted access to a premium offering.
+
+    .. versionadded:: 2.4
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The entitlement's ID.
+    sku_id: :class:`int`
+        The ID of the SKU that the entitlement belongs to.
+    user_id: Optional[:class:`int`]
+        The ID of the user that is granted access to the entitlement.
+    guild_id: Optional[:class:`int`]
+        The ID of the guild that is granted access to the entitlement
+    application_id: :class:`int`
+        The ID of the application that the entitlement belongs to.
+    type: :class:`EntitlementType`
+        The type of the entitlement.
+    consumed: :class:`bool`
+        Whether the entitlement has been consumed. Not applicable to app subscriptions so will typically be ``False``.
+    starts_at: Optional[:class:`datetime.datetime`]
+        A UTC start date which the entitlement is valid. Not present when using test entitlements.
+    ends_at: Optional[:class:`datetime.datetime`]
+        A UTC date which entitlement is no longer valid. Not present when using test entitlements.
+    """
+
+    __slots__ = (
+        '_state',
+        'id',
+        'sku_id',
+        'user_id',
+        'guild_id',
+        'application_id',
+        'type',
+        'consumed',
+        'starts_at',
+        'ends_at',
+    )
+
+    def __init__(self, state: ConnectionState, data: EntitlementPayload):
+        self._state: ConnectionState = state
+        self.id: int = int(data['id'])
+        self.sku_id: int = int(data['sku_id'])
+        self.user_id: Optional[int] = utils._get_as_snowflake(data, 'user_id')
+        self.guild_id: Optional[int] = utils._get_as_snowflake(data, 'guild_id')
+        self.application_id: int = int(data['application_id'])
+        self.type: EntitlementType = try_enum(EntitlementType, data['type'])
+        self.consumed: bool = data.get('consumed', False)
```

Comment:
```suggestion
        self.deleted: bool = data['deleted']
```
See https://github.com/discord/discord-api-docs/pull/6477